.github/workflows/gh-ci-tests.yaml

```diff
@@ -34,6 +34,7 @@ jobs:
         with:
           environment-file: environment.yml
           environment-name: mda-user-guide
+          cache-environment: true
```

Comment:
From a quick look through, it seems like you're getting a cache clash because both actions are using the same key: https://github.com/MDAnalysis/UserGuide/actions/runs/5623009170/job/15237260079#step:3:38

The safest solution here would be to just pass a different `cache-environment-key` for each action. That flatten the differences between the two workflows and put everything in the environment.yml.

Reply:
Actually, thinking about it, flattening will create a simpler situation, where we don't have to use the `cache-environment-key` feature

Reply:
It might be good to do both even if they have the same key. That way we've got extra security if someone does changes in the future (it's one of those "someone new reads the entry it and realises how it all work" extra security steps).

Reply:
Actually @IAlibay I don't think there is a problem with a cache clash: yes the file hash is the same but the args has is different, the output below is from the two runs in the table where the cache was on, the `user-guide-args` is different


```
  Restored cache with key `micromamba-environment--linux-64-mda-user-guide-args-958ca3f-file-69dbdc0bcba969b5795a9da29bcf23f024e7c0672d0bd1630afe58a4cb67706f`
```

```
  Restored cache with key `micromamba-environment--linux-64-mda-user-guide-args-93344c2-file-69dbdc0bcba969b5795a9da29bcf23f024e7c0672d0bd1630afe58a4cb67706f`
```

I'd probably remove the cache key and keep this simpler, this github actions is clearly clever enough. Cherry on the cake we improved the environment.yml